syntax = "proto3";

service LockServer {
    rpc Lock(LockRequest) returns (LockResponse) {}
    rpc Unlock(UnlockRequest) returns (UnlockResponse) {}
    // rpc AddGroup(AddGroupRequest) returns (AddGroupResponse) {}
    rpc AddACL(AddACLRequest) returns (AddACLResponse) {}
    rpc AddFile(AddFileRequest) returns (AddFileResponse) {}
    // rpc DeleteFile(DeleteFileRequest) returns (DeleteFileResponse) {} // this is include when unlocked
}

message LockRequest {
    int32 user_id = 1;
    int32 group_id = 2;
    string file_name = 3;
    int32 lock_type = 4; // 0: read, 1: write 2: delete(delete file when unlocked)
}

message LockResponse {
    bool status = 1;
    string msg = 2;
}

message UnlockRequest {
    int32 user_id = 1;
    int32 group_id = 2;
    string file_name = 3;
    int32 lock_type = 4; // 0: read, 1: write 2: delete(delete file when unlocked)
}

message UnlockResponse {
    bool status = 1;
    string msg = 2;
}

// message AddGroupRequest {
//     int32 group_id = 1;
// }

// message AddGroupResponse {
//     bool status = 1;
// }

message AddACLRequest {
    int32 user_id = 1;
    int32 group_id = 2;
    string file_name = 3;
    bool is_read = 4;
    bool is_write = 5;
    bool is_delete = 6;

}

message AddACLResponse {
    bool status = 1;
    string msg = 2;
}

message AddFileRequest {
    int32 user_id = 1;
    int32 group_id = 2;
    string file_name = 3;
}

message AddFileResponse {
    bool status = 1;
    string msg = 2;
}

